alica_engine/include/engine/planselector/PlanSelector.h

```diff
@@ -42,7 +42,7 @@ class PlanSelector
     const TeamManager& _teamManager;
 
     PlanBase* _pb;
-    const Blackboard* _globalBlackboard;
+    std::atomic<const Blackboard*> _globalBlackboard;
```

Comment:
the global blackboard is only set at initialization i.e. immediately after construction of the PlanSelector, so it this required?

Reply:
```
==================
[1m[31mWARNING: ThreadSanitizer: data race (pid=961383)
[1m[0m[1m[34m  Read of size 8 at 0x7b18000014f0 by thread T15 (mutexes: write M325802165941859224):
[1m[0m    #0 alica::PlanSelector::createRunningPlan(alica::RunningPlan*, std::vector<alica::Plan const*, std::allocator<alica::Plan const*> > const&, alica::Configuration const*, std::vector<unsigned long, std::allocator<unsigned long> > const&, alica::RunningPlan const*, alica::PlanType const*, double&) alica/alica_engine/src/engine/planselector/PlanSelector.cpp:114 (libalica_engine.so+0x58c0d2)
    #1 alica::PlanSelector::getPlansForStateInternal(alica::RunningPlan*, std::vector<alica::ConfAbstractPlanWrapper const*, std::allocator<alica::ConfAbstractPlanWrapper const*> > const&, std::vector<unsigned long, std::allocator<unsigned long> > const&, std::vector<alica::RunningPlan*, std::allocator<alica::RunningPlan*> >&) alica/alica_engine/src/engine/planselector/PlanSelector.cpp:214 (libalica_engine.so+0x58cbf2)
    #2 alica::PlanSelector::getPlansForState(alica::RunningPlan*, std::vector<alica::ConfAbstractPlanWrapper const*, std::allocator<alica::ConfAbstractPlanWrapper const*> > const&, std::vector<unsigned long, std::allocator<unsigned long> > const&, std::vector<alica::RunningPlan*, std::allocator<alica::RunningPlan*> >&) alica/alica_engine/src/engine/planselector/PlanSelector.cpp:87 (libalica_engine.so+0x58bbf8)
    #3 alica::RuleBook::allocationRule(alica::RunningPlan&) alica/alica_engine/src/engine/RuleBook.cpp:362 (libalica_engine.so+0x3eac2f)
    #4 alica::RuleBook::visit(alica::RunningPlan&) alica/alica_engine/src/engine/RuleBook.cpp:112 (libalica_engine.so+0x3e8dce)
    #5 alica::RunningPlan::tick(alica::RuleBook*) alica/alica_engine/src/engine/RunningPlan.cpp:176 (libalica_engine.so+0x40d677)
    #6 alica::PlanBase::run(alica::Plan const*) alica/alica_engine/src/engine/PlanBase.cpp:180 (libalica_engine.so+0x3c6de4)
    #7 void std::__invoke_impl<void, void (alica::PlanBase::*)(alica::Plan const*), alica::PlanBase*, alica::Plan const*>(std::__invoke_memfun_deref, void (alica::PlanBase::*&&)(alica::Plan const*), alica::PlanBase*&&, alica::Plan const*&&) /usr/include/c++/9/bits/invoke.h:73 (libalica_engine.so+0x3d8262)
    #8 std::__invoke_result<void (alica::PlanBase::*)(alica::Plan const*), alica::PlanBase*, alica::Plan const*>::type std::__invoke<void (alica::PlanBase::*)(alica::Plan const*), alica::PlanBase*, alica::Plan const*>(void (alica::PlanBase::*&&)(alica::Plan const*), alica::PlanBase*&&, alica::Plan const*&&) /usr/include/c++/9/bits/invoke.h:95 (libalica_engine.so+0x3d8095)
    #9 void std::thread::_Invoker<std::tuple<void (alica::PlanBase::*)(alica::Plan const*), alica::PlanBase*, alica::Plan const*> >::_M_invoke<0ul, 1ul, 2ul>(std::_Index_tuple<0ul, 1ul, 2ul>) /usr/include/c++/9/thread:244 (libalica_engine.so+0x3d7f30)
    #10 std::thread::_Invoker<std::tuple<void (alica::PlanBase::*)(alica::Plan const*), alica::PlanBase*, alica::Plan const*> >::operator()() /usr/include/c++/9/thread:251 (libalica_engine.so+0x3d7e9a)
    #11 std::thread::_State_impl<std::thread::_Invoker<std::tuple<void (alica::PlanBase::*)(alica::Plan const*), alica::PlanBase*, alica::Plan const*> > >::_M_run() /usr/include/c++/9/thread:195 (libalica_engine.so+0x3d7e4c)
    #12 <null> <null> (libstdc++.so.6+0xd6de3)

[1m[34m  Previous write of size 8 at 0x7b18000014f0 by main thread:
[1m[0m    #0 alica::PlanSelector::setGlobalBlackboard(alica::Blackboard const*) alica/alica_engine/src/engine/planselector/PlanSelector.cpp:235 (libalica_engine.so+0x58d02e)
    #1 alica::RuleBook::init(alica::Blackboard const*) alica/alica_engine/src/engine/RuleBook.cpp:52 (libalica_engine.so+0x3e881a)
    #2 alica::PlanBase::start(alica::Plan const*) alica/alica_engine/src/engine/PlanBase.cpp:122 (libalica_engine.so+0x3c6687)
    #3 alica::AlicaEngine::start() alica/alica_engine/src/engine/AlicaEngine.cpp:114 (libalica_engine.so+0x3851db)
    #4 TestBody alica/supplementary/alica_ros1/alica_tests/src/test/test_adjacent_plans_success.cpp:27 (alica_tests-test+0x1dcfb0)
    #5 void testing::internal::HandleSehExceptionsInMethodIfSupported<testing::Test, void>(testing::Test*, void (testing::Test::*)(), char const*) /usr/src/googletest/googletest/src/gtest.cc:2433 (libgtestd.so+0xcf04f)
    #6 void testing::internal::HandleExceptionsInMethodIfSupported<testing::Test, void>(testing::Test*, void (testing::Test::*)(), char const*) /usr/src/googletest/googletest/src/gtest.cc:2469 (libgtestd.so+0xc3a69)
    #7 testing::Test::Run() /usr/src/googletest/googletest/src/gtest.cc:2508 (libgtestd.so+0x938d5)
    #8 testing::TestInfo::Run() /usr/src/googletest/googletest/src/gtest.cc:2684 (libgtestd.so+0x94590)
    #9 testing::TestSuite::Run() /usr/src/googletest/googletest/src/gtest.cc:2816 (libgtestd.so+0x9506d)
    #10 testing::internal::UnitTestImpl::RunAllTests() /usr/src/googletest/googletest/src/gtest.cc:5338 (libgtestd.so+0xa34b2)
    #11 bool testing::internal::HandleSehExceptionsInMethodIfSupported<testing::internal::UnitTestImpl, bool>(testing::internal::UnitTestImpl*, bool (testing::internal::UnitTestImpl::*)(), char const*) /usr/src/googletest/googletest/src/gtest.cc:2433 (libgtestd.so+0xd0fb9)
    #12 bool testing::internal::HandleExceptionsInMethodIfSupported<testing::internal::UnitTestImpl, bool>(testing::internal::UnitTestImpl*, bool (testing::internal::UnitTestImpl::*)(), char const*) /usr/src/googletest/googletest/src/gtest.cc:2469 (libgtestd.so+0xc56ab)
    #13 testing::UnitTest::Run() /usr/src/googletest/googletest/src/gtest.cc:4925 (libgtestd.so+0xa15af)
    #14 RUN_ALL_TESTS() /usr/src/googletest/googletest/include/gtest/gtest.h:2473 (alica_tests-test+0xfffc2)
    #15 main alica/supplementary/alica_ros1/alica_tests/src/test/test_main.cpp:10 (alica_tests-test+0xffe71)
```